src/appforms/tests/tests/backbone/040-view02field14field_map.js

```diff
@@ -0,0 +1,163 @@
+var assert = chai.assert;
+
+
+describe("Backbone Map View", function() {
+
+  var oldGeolocation = navigator.geolocation;
+
+  var mapForm = {
+    "_id": "58a6d7f8ed4560ba675e31c7",
+    "createdBy": "ndonnell@redhat.com",
+    "description": "Blank form with no fields",
+    "name": "testniallmap",
+    "updatedBy": "ndonnell@redhat.com",
+    "dataTargets": [],
+    "dataSources": {"formDataSources": []},
+    "subscribers": [],
+    "pageRules": [],
+    "fieldRules": [],
+    "pages": [{
+      "_id": "58a6d7f8ed4560ba675e31c6",
+      "fields": [{
+        "required": true,
+        "type": "locationMap",
+        "name": "Map",
+        "fieldCode": null,
+        "_id": "58a6d800b73a4cb067d73809",
+        "adminOnly": false,
+        "fieldOptions": {"validation": {"validateImmediately": true}},
+        "repeating": false,
+        "dataSourceType": "static"
+      }]
+    }],
+    "lastUpdated": "2017-02-17T11:01:20.600Z",
+    "dateCreated": "2017-02-17T11:01:12.419Z",
+    "lastDataRefresh": "2017-02-17T11:01:20.600Z",
+    "pageRef": {"58a6d7f8ed4560ba675e31c6": 0},
+    "fieldRef": {"58a6d800b73a4cb067d73809": {"page": 0, "field": 0}},
+    "lastUpdatedTimestamp": 1487329280600
+  };
+
+  var mockError = {
+    code: "PERMISSION_DENIED",
+    message: "Map Permission Denied"
+  };
+
+  var changeGeolocation = function(object) {
```

Comment:
Do we have access to something like sinon.js in these tests? It would definitely reduce the boilerplate around this kind of mocking.